plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/partition/HudiPartitionInfoLoader.java

```diff
@@ -64,7 +64,7 @@ private void generateSplitsFromPartition(String partitionName)
     {
         Optional<HudiPartitionInfo> partitionInfo = hudiDirectoryLister.getPartitionInfo(partitionName);
         partitionInfo.ifPresent(hudiPartitionInfo -> {
-            if (hudiPartitionInfo.doesMatchPredicates() || partitionName.equals("")) {
+            if (hudiPartitionInfo.doesMatchPredicates() || partitionName.isEmpty()) {
```

Comment:
An empty `partitionName` means the table isn't partitioned, right? Is this condition really reachable?

Reply:
In my understanding is yes cc @codope

Reply:
See here, for non-partitioned tables https://github.com/trinodb/trino/blob/6a0e30fe1a99157c321055c35275b66b975900aa/plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java#L111

Reply:
This isn't intuitive. I would recommend leaving a code comment. 

Also, the empty string comes from the following place. It would be nice to extract a contract (e.g. `EMPTY_PARTITION`) and use it instead of `isEmpty()`. 
https://github.com/trinodb/trino/blob/6a0e30fe1a99157c321055c35275b66b975900aa/plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java#L108-L112